Userland/Libraries/LibMarkdown/Text.h

```diff
@@ -1,54 +1,129 @@
 /*
  * Copyright (c) 2019-2020, Sergey Bugaev <bugaevc@serenityos.org>
+ * Copyright (c) 2021, Peter Elliott <pelliott@serenityos.org>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
 #pragma once
 
 #include <AK/Noncopyable.h>
+#include <AK/NonnullOwnPtrVector.h>
+#include <AK/OwnPtr.h>
 #include <AK/String.h>
-#include <AK/Vector.h>
 
 namespace Markdown {
 
 class Text final {
-    AK_MAKE_NONCOPYABLE(Text);
-
 public:
-    struct Style {
-        bool emph { false };
-        bool strong { false };
-        bool code { false };
-        String href;
-        String img;
+    struct Node {
+        virtual void render_to_html(StringBuilder& builder) const = 0;
+        virtual void render_for_terminal(StringBuilder& builder) const = 0;
+        virtual size_t length() const = 0;
+
+        virtual ~Node() { }
+    };
+
+    struct Emph : Node {
```

Comment:
Is this name from a spec? ideally we'd use a non-abbreviated name for this (e.g. `Emphasis`)
if it _is_ from a spec, could we link to it on things that are strictly markdown concepts? (for instance, "flanking" and "delimiter runs" don't make much sense to me without any context)

Reply:
'Emphasis' is the term from the spec, but 'Emph' was what was used in the parser before. I can rename these. I'm not sure that I understand your second point. Do you want comments with references to the spec when a new term is introduced?

Reply:
Only on things that really don't make much sense without either an inline explanation of the concept or a link to the spec where the term is introduced.
I'm not asking for spec-linkification, just explanation on the terms or concepts (I believe the two I gave for an example are the only ones that really needed that)

Reply:
`Emph` has been renamed `EmphasisNode`, and comments on delimiter runs and flanking have been added.